pkg/client/client.go

```diff
@@ -12,7 +12,12 @@ import (
 type SecretsClient interface {
 	SetSecret(ctx context.Context, storeName, id string, request *types.SetSecretRequest) (*types.SecretResponse, error)
 	GetSecret(ctx context.Context, storeName, id, version string) (*types.SecretResponse, error)
+	GetDeletedSecret(ctx context.Context, storeName, id string) (*types.SecretResponse, error)
```

Comment:
`GetDeleted` and `ListDeleted` are just Get and List but with a query param. This is the example for Eth1 in the handler:
```
getDeleted := request.URL.Query().Get("deleted")
	var eth1Acc *entities.ETH1Account
	if getDeleted == "" {
		eth1Acc, err = eth1Store.Get(ctx, getAddress(request))
	} else {
		eth1Acc, err = eth1Store.GetDeleted(ctx, getAddress(request))
	}
```

Reply:
Applying same format